### PR TITLE
Fix for Test Failures : CEPH-83584070, CEPH-83587957

### PR DESCRIPTION
**Pull request by chaitanyadommeti**
Added long_running as the default timeout is changed to 300sec in ceph.py recently

http://magna002.ceph.redhat.com/cephci-jenkins/results/openstack/RH/7.1/rhel-9/Weekly/18.2.1-262/rbd/58/tier-3_rbd_migration/Test_image_migration_with_external_qcow_data_format_0.log

 the qcow image in curl -o /mnt/ubuntu-12.04.qcow2 http://download.ceph.com/qa/ubuntu-12.04.qcow2 is of size 1.2 gb and hence it takes quite an amount of time to download. Hence used long_running option as the timeout for long_running is 1hr.

Removed the dependency of .env file as of now in rest/common/utils/rest.py:
http://magna002.ceph.redhat.com/cephci-jenkins/results/openstack/RH/8.0/rhel-9/Weekly/19.2.0-52/rbd/12/tier-2_rbd_rest/Run_rest_API_tests_to_create_pool_and_image_0.log

Passed the arguments from the test file to  rest.py to fetch the cluster object and obtain IP.
Added parameters 'initial-dashboard-password' and 'dashboard-password-noupdate' to the suites file to utilize the default credentials that exists in rest.py.

### CODE REVIEW

File: rest/common/utils/rest.py [hunk 1]
```diff
@@ -6,26 +6,32 @@
 import time
 
 import requests
-from dotenv import dotenv_values
 
 from rest.common.config.config import Config
 from rest.common.utils.exceptions import CommandExecutionError, HTTPError
 from utility.log import Log
 
+# from dotenv import dotenv_values
```
Comment (sunilangadi2) [inline rest/common/utils/rest.py:14]: please uncomment this?

File: rest/common/utils/rest.py [hunk 2]
```diff
@@ -6,26 +6,32 @@
...
     # config = {"IP": "10.12.34.23", "USERNAME": "admin", "PASSWORD":"passwd", "PORT":"3211"}
     """
-    config = dotenv_values(".env")
-    username = config["USERNAME"]
-    password = config["PASSWORD"]
-    port = config["PORT"]
-    ip = config["IP"]
-    rest = REST(ip=ip, username=username, password=password, port=port)
+    # config = dotenv_values(".env")
+    # username = config["USERNAME"]
```
Comment (sunilangadi2) [inline rest/common/utils/rest.py:26]: same here

**Review by sunilangadi2 — CHANGES_REQUESTED** [review]
please add successful test results.

File: rest/common/utils/rest.py [hunk 3]
```diff
@@ -6,26 +6,32 @@
...
+    # password = config["PASSWORD"]
+    # port = config["PORT"]
+    # ip = config["IP"]
+    # rest = REST(ip=ip, username=username, password=password, port=port)
+    ceph_cluster = kw.get("ceph_cluster")
```
Comment (HaruChebrolu) [inline rest/common/utils/rest.py:31]: Let it throw exception if we don't have ceph_cluster in kw.
ceph_cluster = kw["ceph_cluster"]

File: rest/common/utils/rest.py (hunk 2 above)
Comment (chaitanyadommeti) [reply rest/common/utils/rest.py:26]: Test log: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-QQQCNY/
Note: test failed due to other reason but the rest api call was successful for which the fix was provided here.

File: rest/common/utils/rest.py (hunk 3 above)
Comment (chaitanyadommeti) [reply rest/common/utils/rest.py]: Addressed

File: rest/common/utils/rest.py (hunk 2 above)
Comment (sunilangadi2) [reply rest/common/utils/rest.py:26]: ok but please get success logs

**Comment by sunilangadi2** [comment]
@chaitanyadommeti please attach success logs.

File: rest/common/utils/rest.py [hunk 4]
```diff
@@ -6,26 +6,32 @@
...
 from utility.log import Log
 
+# from dotenv import dotenv_values
+
+
 log = Log(__name__)
 
 
-def rest():
+def rest(**kw):
```
Comment (aarsharm) [inline rest/common/utils/rest.py:20]: is this parameter optional?
We are using this rest.py in other NVME scripts also where no arguments are passed while calling the function.
Will it not regress them?

Some other files where this has been used:
1) [tests/nvmeof/test_nvmeof_rest_sanity.py](https://github.com/red-hat-storage/cephci/blob/2dd8e6480224663af1697ba62b61e702bb49191c/tests/nvmeof/test_nvmeof_rest_sanity.py#L27)
2) [tests/nvmeof/rest/test_nvmeof_rest_sanity.py](https://github.com/red-hat-storage/cephci/blob/2dd8e6480224663af1697ba62b61e702bb49191c/tests/nvmeof/rest/test_nvmeof_rest_sanity.py#L25)

File: tests/rbd/test_rbd_image_migration_qcow.py [hunk 5]
```diff
@@ -92,13 +93,15 @@ def test_qcow_image_migration(rbd_obj, client, **kw):
...
                 client_node=client,
+                long_running=True,
                 **kw,
             )
 
             # verify execute migration status
             if verify_migration_state(
                 action="execute",
                 image_spec=image_spec,
+                long_running=True,
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_image_migration_qcow.py:104]: @chaitanyadommeti add `long running` for all three actions such migration prepare, execute, commit

**Review by sunilangadi2 — CHANGES_REQUESTED** [review]

**Comment by chaitanyadommeti** [comment]
Closing this PR. Raised following PRs separately for the two TFA failures addressed here. 
https://github.com/red-hat-storage/cephci/pull/4297
https://github.com/red-hat-storage/cephci/pull/4298

**Pull request closed without merging**